src/OlFlatStyleParser.ts

```diff
@@ -0,0 +1,297 @@
+  }
+
+  flatStyleToGeoStylerTextSymbolizer(flatStyle: FlatStyle): TextSymbolizer {
+    const [textFillColor, textFillOpacity] = OlFlatStyleUtil.isExpression(flatStyle['text-fill-color'])
+      ? [OlFlatStyleUtil.olExpressionToGsExpression(flatStyle['text-fill-color']) as Expression<string>]
+      : OlFlatStyleUtil.getColorAndOpacity(flatStyle['text-fill-color']);
+
+    const [textStrokeColor, textStrokeOpacity] = OlFlatStyleUtil.isExpression(flatStyle['text-stroke-color'])
+      ? [OlFlatStyleUtil.olExpressionToGsExpression(flatStyle['text-stroke-color']) as Expression<string>]
+      : OlFlatStyleUtil.getColorAndOpacity(flatStyle['text-stroke-color']);
+
+    // NOTE: If font is an expression, we cannot detect the size
+    const [font, fontSize] = OlFlatStyleUtil.isExpression(flatStyle['text-font'])
+      ? [[OlFlatStyleUtil.olExpressionToGsExpression(flatStyle['text-font']) as Expression<string>], undefined]
+      : flatStyle['text-font'] ? [
+        [OlStyleUtil.getFontNameFromOlFont(flatStyle['text-font'])],
+        OlStyleUtil.getSizeFromOlFont(flatStyle['text-font'])
+      ] : [];
+
+    // TODO add missing properties
+    return {
+      kind: 'Text',
+      label: OlFlatStyleUtil.olExpressionToGsExpression(flatStyle['text-value']) as Expression<string>,
+      color: textFillColor,
+      opacity: textFillOpacity,
+      haloColor: textStrokeColor,
+      haloOpacity: textStrokeOpacity,
+      haloWidth: OlFlatStyleUtil.olExpressionToGsExpression(flatStyle['text-stroke-width']) as Expression<number>,
+      font,
+      size: fontSize,
+      maxAngle: OlFlatStyleUtil.olExpressionToGsExpression(flatStyle['text-max-angle']) as Expression<number>,
+      offset: [
+        OlFlatStyleUtil.olExpressionToGsExpression(flatStyle['text-offset-x']) as Expression<number>,
+        OlFlatStyleUtil.olExpressionToGsExpression(flatStyle['text-offset-y']) as Expression<number>,
+      ],
+      // TODO check if values of ol placement are the same as values of geostyler placement
+      placement: OlFlatStyleUtil.olExpressionToGsExpression(
+        flatStyle['text-placement']
+      ) as Expression<'point' | 'line' | 'line-center'>,
+      rotate: OlFlatStyleUtil.olExpressionToGsExpression(flatStyle['text-rotation']) as Expression<number>,
+      justify: OlFlatStyleUtil.olExpressionToGsExpression(
+        flatStyle['text-justify']
+      ) as Expression<'left' | 'center' | 'right'>,
+      padding: OlFlatStyleUtil.olExpressionToGsExpression(flatStyle['text-padding']) as Expression<number>,
+    };
+  }
+
+  flatStyleToGeoStylerIconSymbolizer(flatStyle: FlatStyle): IconSymbolizer {
+    // TODO add missing properties
+    return {
+      kind: 'Icon',
+      image: flatStyle['icon-src'],
+      offset: OlFlatStyleUtil.olExpressionToGsExpression(
+        flatStyle['icon-offset']
+      ) as [Expression<number>, Expression<number>],
+      opacity: OlFlatStyleUtil.olExpressionToGsExpression(flatStyle['icon-opacity']) as Expression<number>,
```

Comment:
You're using generics for `olExpressionToGsExpression` but also `as` statements after every call. Maybe you can improve the generics to get rid of the `as` or remove the generics code from `olExpressionToGsExpression`.

Reply:
done